templates.go

```diff
@@ -600,62 +600,77 @@ coreos:
       ExecStartPre=/bin/bash -c "while [ ! -f /etc/kubernetes/ssl/etcd/server-crt.pem ]; do echo 'Waiting for /etc/kubernetes/ssl/etcd/server-crt.pem to be written' && sleep 1; done"
       ExecStartPre=/bin/bash -c "while [ ! -f /etc/kubernetes/ssl/etcd/server-key.pem ]; do echo 'Waiting for /etc/kubernetes/ssl/etcd/server-key.pem to be written' && sleep 1; done"
-      ExecStart=/usr/bin/etcd2 --advertise-client-urls=https://{{ .Cluster.Etcd.Domain }}:2379 \
-                               --data-dir=/etc/kubernetes/data/etcd/ \
-                               --initial-advertise-peer-urls=https://127.0.0.1:2380 \
-                               --listen-client-urls=https://0.0.0.0:2379 \
-                               --listen-peer-urls=https://${DEFAULT_IPV4}:2380 \
-                               --initial-cluster-token k8s-etcd-cluster \
-                               --initial-cluster etcd0=https://127.0.0.1:2380 \
-                               --initial-cluster-state new \
-                               --ca-file=/etc/kubernetes/ssl/etcd/server-ca.pem \
-                               --cert-file=/etc/kubernetes/ssl/etcd/server-crt.pem \
-                               --key-file=/etc/kubernetes/ssl/etcd/server-key.pem \
-                               --peer-ca-file=/etc/kubernetes/ssl/etcd/server-ca.pem \
-                               --peer-cert-file=/etc/kubernetes/ssl/etcd/server-crt.pem \
-                               --peer-key-file=/etc/kubernetes/ssl/etcd/server-key.pem \
-                               --peer-client-cert-auth=true \
-                               --name etcd0
+      ExecStart=/usr/bin/docker run \
+          -v /etc/ssl/certs/ca-certificates.crt:/etc/ssl/certs/ca-certificates.crt \
+          -v /etc/kubernetes/ssl/etcd/:/etc/etcd \
+          -v /etc/kubernetes/data/etcd/:/var/lib/etcd  \
+          --net=host  \
+          --name $NAME \
+          $IMAGE \
+          etcd \
+          --name etcd0 \
+          --trusted-ca-file /etc/etcd/server-ca.pem \
+          --cert-file /etc/etcd/server-crt.pem \
+          --key-file /etc/etcd/server-key.pem\
+          --peer-trusted-ca-file /etc/etcd/server-ca.pem \
+          --peer-cert-file /etc/etcd/server-crt.pem \
+          --peer-key-file /etc/etcd/server-key.pem \
+          --peer-client-cert-auth=true \
+          --advertise-client-urls=https://{{ .Cluster.Etcd.Domain }}:2379 \
+          --initial-advertise-peer-urls=https://127.0.0.1:2380 \
+          --listen-client-urls=https://0.0.0.0:2379 \
+          --listen-peer-urls=https://${DEFAULT_IPV4}:2380 \
+          --initial-cluster-token k8s-etcd-cluster \
+          --initial-cluster etcd0=https://127.0.0.1:2380 \
+          --initial-cluster-state new \
+          --data-dir=/var/lib/etcd
 
       [Install]
       WantedBy=multi-user.target
-  - name: etcd2-restart.service
+  - name: etcd3-restart.service
```

Comment:
I didn't query it because several other units have a restart service. But I don't know why its there so I should have done :(

I'm fine to remove it if its not needed. For restarts currently masters are single node. But I'd like to see etcd move to the host cluster, which I think is the plan?